packages/plugin-commands-publishing/test/publish.ts

```diff
@@ -277,6 +277,55 @@ test('publish: package with all possible fields in publishConfig', async () => {
   })
 })
 
+test('publish: package with publishConfig.directory', async () => {
+  const packages = preparePackages([
+    {
+      name: 'test-publish-config-directory',
+      version: '1.0.0',
+
+      bin: './bin.js',
+      main: './index.js',
+      module: './index.mjs',
+      types: './types.d.ts',
+      typings: './typings.d.ts',
+
+      publishConfig: {
+        directory: 'dist',
+      },
+    },
+  ])
+
+  const testPublishConfigDirectory = packages['test-publish-config-directory']
+
+  expect(testPublishConfigDirectory).toBeTruthy()
+
+  await fs.mkdir(path.join(testPublishConfigDirectory.dir(), 'dist'))
+
+  await fs.writeFile(
+    path.join(testPublishConfigDirectory.dir(), 'dist/package.json'),
+    `
+    {
+      "name": "publish_config_directory_dist_package",
+      "version": "1.0.0"
+    }
+  `,
+    {
+      encoding: 'utf-8',
+    }
+  )
+
+  process.chdir('test-publish-config-directory')
+
+  await publish.handler(
+    {
+      ...DEFAULT_OPTS,
+      argv: { original: ['publish', ...CREDENTIALS] },
+      dir: process.cwd(),
+    },
+    []
+  )
+})
```

Comment:
Here is an example how to install the published package to verify it

https://github.com/pnpm/pnpm/blob/f4aac60483ca2616549eadffd7871e36f77b3e0e/packages/plugin-commands-publishing/test/publish.ts#L128-L175

Reply:
thank you, simply adding the package worked, previously I was trying with a different new package and calling install, didn't work for some reason 🤷‍♂️